keyboards/winry/winry25tc/keymaps/lightsout/keymap.c

```diff
@@ -0,0 +1,165 @@
+/* Copyright 2023 Tom Parker-Shemilt <palfrey@tevp.net>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, version 3 of the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdlib.h>
+
+#include QMK_KEYBOARD_H
+
+// clang-format off
+enum my_keycodes {
+    K00 = SAFE_RANGE,
+    K01,
+    K02,
+    K03,
+    K04,
+    K05,
+    K06,
+    K07,
+    K08,
+    K09,
+    K10,
+    K11,
+    K12,
+    K13,
+    K14,
+    K15,
+    K16,
+    K17,
+    K18,
+    K19,
+    K20,
+    K21,
+    K22,
+    K23,
+    K24
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT(
+        K00, K01, K02, K03, K04,
+        K05, K06, K07, K08, K09,
+        K10, K11, K12, K13, K14,
+        K15, K16, K17, K18, K19,
+        K20, K21, K22, K23, K24
+    )
+};
+
+uint8_t tiles[5][5] = {
```

Comment:
you could probably use bool for this.  Or if you wanted to be fancy, bitmasks

Reply:
I've used bools as bitmasks feels like overkill given there's still plenty of space on the device. One for someone else if they want to port it to something smaller :)